mixed/spark/common/src/main/java/com/netease/arctic/spark/SparkUnifiedCatalog.java

```diff
@@ -0,0 +1,338 @@
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+import org.apache.spark.util.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+public class SparkUnifiedCatalog implements TableCatalog, SupportsNamespaces, ProcedureCatalog {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkUnifiedCatalog.class);
+  private static final Map<TableFormat, String> defaultTableCatalogImplMap =
+      ImmutableMap.of(
+          TableFormat.ICEBERG, "org.apache.iceberg.spark.SparkCatalog",
+          TableFormat.MIXED_HIVE, "com.netease.arctic.spark.ArcticSparkCatalog",
+          TableFormat.MIXED_ICEBERG, "com.netease.arctic.spark.ArcticSparkCatalog",
+          TableFormat.PAIMON, "org.apache.paimon.spark.SparkCatalog");
+
+  private UnifiedCatalog unifiedCatalog;
+  private String name;
+  private final Map<TableFormat, SparkTableFormat> tableFormats = Maps.newConcurrentMap();
+  private final Map<TableFormat, TableCatalog> tableCatalogs = Maps.newConcurrentMap();
+
+  @Override
+  public void initialize(String name, CaseInsensitiveStringMap options) {
+    Map<String, String> properties = Maps.newHashMap(options);
+    String uri = options.get(SparkUnifiedCatalogProperties.URI);
+    properties.remove(SparkUnifiedCatalogProperties.URI);
+    Preconditions.checkNotNull(uri, "lack required option: %s", SparkUnifiedCatalogProperties.URI);
+
+    ArcticThriftUrl catalogUri = ArcticThriftUrl.parse(uri, Constants.THRIFT_TABLE_SERVICE_NAME);
+    String registerCatalogName = catalogUri.catalogName();
+
+    if (StringUtils.isBlank(registerCatalogName)) {
+      registerCatalogName = name;
+      if (CatalogManager.SESSION_CATALOG_NAME().equalsIgnoreCase(registerCatalogName)) {
+        LOG.warn(
+            "Catalog name is not exists in catalog uri, using spark catalog as register catalog name, but "
+                + "current name "
+                + registerCatalogName
+                + " is spark session catalog name.");
+      }
+    }
+    this.name = name;
+    this.unifiedCatalog =
+        UnifiedCatalogLoader.loadUnifiedCatalog(
+            catalogUri.serverUrl(), registerCatalogName, properties);
+    ServiceLoader<SparkTableFormat> sparkTableFormats = ServiceLoader.load(SparkTableFormat.class);
+    for (SparkTableFormat format : sparkTableFormats) {
+      tableFormats.put(format.format(), format);
+    }
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  private String namespaceToDatabase(String[] namespace) {
+    Preconditions.checkArgument(namespace.length == 1, "only support namespace with 1 level.");
+    return namespace[0];
+  }
+
+  @Override
+  public String[][] listNamespaces() throws NoSuchNamespaceException {
```

Comment:
We should remove unnecessary throw declarations.

Reply:
I found there are other same issues in this class, can you fix them all?

Reply:
fixed